### PR TITLE
feature: add "volume" flag when remove container

### PR DESCRIPTION
**Pull request by rudyfly**
### Ⅰ. Describe what this PR did
Add "volume" flag when remove container, it is used to remove these
volumes that is created by container.

### Ⅱ. Does this pull request fix one issue?

NONE

### Ⅳ. Describe how to verify it
1. create volume
```
# pouch volume create -n rmVolume-test-volume
Scope:
Status:       map[mount:/data/pouch/volume sifter:Default]
CreatedAt:    2018-4-28 15:27:28
Driver:       local
Labels:       map[]
Mountpoint:   /data/pouch/volume/rmVolume-test-volume
Name:         rmVolume-test-volume
```
2. run container with volume
```
# pouch run --name rmVolume-test -d -v rmVolume-test-volume:/mnt -v /home  busybox:latest top
```
3. check volumes
```
# pouch volume ls
DRIVER   VOLUME NAME
local    rmVolume-test-volume
local    2b435c1fada0cf10064f3e9e39de3b82737bbdd6207d6f4f1e9e830ee76f2812
```
4. remove container with `-v`
```
# pouch rm -vf rmVolume-test
```
5. check the volume exist
```
# pouch volume ls
DRIVER   VOLUME NAME
local    rmVolume-test-volume
```
### Ⅴ. Special notes for reviews

Signed-off-by: Rudy Zhang <rudyflyzhang@gmail.com>

### CODE REVIEW

File: cli/rm.go [hunk 1]
```diff
@@ -39,16 +42,24 @@ func (r *RmCommand) Init(c *Cli) {
 
 // addFlags adds flags for specific command.
 func (r *RmCommand) addFlags() {
-	r.cmd.Flags().BoolVarP(&r.force, "force", "f", false, "if the container is running, force to remove it")
+	flagSet := r.cmd.Flags()
+
+	flagSet.BoolVarP(&r.force, "force", "f", false, "if the container is running, force to remove it")
+	flagSet.BoolVarP(&r.removeVolume, "volumes", "v", false, "remove container's volumes that create by the container")
```
Comment (fuweid) [inline cli/rm.go]: it's better to use `removeVolume` instead of `removeVolumes`

File: cli/run.go [hunk 2]
```diff
@@ -7,6 +7,7 @@ import (
 	"os"
 	"strings"
 
+	"github.com/alibaba/pouch/apis/types"
```
Comment (fuweid) [inline cli/run.go]: add new blank line here

File: daemon/mgr/container.go [hunk 3]
```diff
@@ -2271,6 +2271,12 @@ func (mgr *ContainerManager) detachVolumes(ctx context.Context, c *ContainerMeta
 		}
 
 		mgr.VolumeMgr.Detach(ctx, name, option)
```
Comment (HusterWan) [inline daemon/mgr/container.go]: `Detach` is to `umount` volume mountpoint, so if the volume is attached to multi containers, does this action will cause other containers' volumes unused

File: test/environment/cleanup.go [hunk 4]
```diff
@@ -4,6 +4,7 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/alibaba/pouch/apis/types"
 	"github.com/alibaba/pouch/client"
 	"github.com/pkg/errors"
```
Comment (fuweid) [inline test/environment/cleanup.go]: add new blank line here

**Comment by rudyfly** [comment]
PTAL @HusterWan @fuweid

File: client/container_remove_test.go [hunk 5]
```diff
@@ -24,7 +26,7 @@ func TestContainerRemoveNotFoundError(t *testing.T) {
 	client := &APIClient{
 		HTTPCli: newMockClient(errorMockResponse(http.StatusNotFound, "Not Found")),
 	}
-	err := client.ContainerRemove(context.Background(), "no contaienr", true)
+	err := client.ContainerRemove(context.Background(), "no contaienr", &types.ContainerRemoveOptions{Force: true})
```
Comment (HusterWan) [inline client/container_remove_test.go]: %s/contaienr/container :)

File: daemon/mgr/volume.go [hunk 6]
```diff
@@ -147,6 +157,26 @@ func (vm *VolumeManager) Attach(ctx context.Context, name string, options map[st
...
+	if err != nil {
+		return nil, errors.Errorf("failed to get volume: %s", name)
+	}
+
+	if options == nil {
+		options = make(map[string]string)
+	}
+
+	cid, ok := options[types.OptionRef]
+	if ok && len(cid) != 0 {
```
Comment (HusterWan) [inline daemon/mgr/volume.go]: cid == "" will be better WDYT???

Comment (fuweid) [reply daemon/mgr/volume.go]: @HusterWan I think two kinds are the same. Just keep the consistent in one project.

**Comment by HusterWan** [comment]
LGTM

**Pull request merged**